lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -227,7 +232,8 @@ defmodule Mix.Compilers.Elixir do
         runtime_references: runtime_references,
         compile_dispatches: compile_dispatches,
         runtime_dispatches: runtime_dispatches,
-        external: source_external
+        external: source_external,
+        warnings: source(source_entry, :warnings)
```

Comment:
Nevermind, changing the Agent's state to have maps as a warning takes care of it.